skrifa/src/outline/hint.rs

```diff
@@ -4,11 +4,179 @@ use super::{
     cff,
     glyf::{self, FreeTypeScaler},
     pen::PathStyle,
-    AdjustedMetrics, DrawError, Hinting, LocationRef, NormalizedCoord, OutlineCollectionKind,
-    OutlineGlyph, OutlineGlyphCollection, OutlineKind, OutlinePen, Size,
+    AdjustedMetrics, DrawError, GlyphStyles, Hinting, LocationRef, NormalizedCoord,
+    OutlineCollectionKind, OutlineGlyph, OutlineGlyphCollection, OutlineKind, OutlinePen, Size,
 };
 use crate::alloc::{boxed::Box, vec::Vec};
 
+/// Configuration settings for a hinting instance.
+#[derive(Clone, Default, Debug)]
+pub struct HintingOptions {
+    /// Specifies the hinting engine to use.
+    ///
+    /// Defaults to [`Engine::AutoFallback`].
+    pub engine: Engine,
+    /// Defines the properties of the intended target of a hinted outline.
+    ///
+    /// Defaults to a target with [`SmoothMode::Normal`] which is equivalent
+    /// to `FT_RENDER_MODE_NORMAL` in FreeType.
+    pub target: Target,
+}
+
+impl From<Target> for HintingOptions {
+    fn from(value: Target) -> Self {
+        Self {
+            engine: Engine::AutoFallback,
+            target: value,
+        }
+    }
+}
+
+/// Specifies the backend to use when applying hints.
+#[derive(Clone, Default, Debug)]
+pub enum Engine {
+    /// The TrueType or PostScript interpreter.
+    Interpreter,
+    /// The automatic hinter that performs just-in-time adjustment of
+    /// outlines.
+    ///
+    /// Glyph styles can be precomputed per font and may be provided here
+    /// as an optimization to avoid recomputing them for each instance.
+    Auto(Option<GlyphStyles>),
+    /// Selects the engine based on the same rules that FreeType uses in
+    /// the default configuration.
```

Comment:
AFAICS FreeType uses load flags to control hinting behavior. What do you mean by "default configuration" here, do you mean no extra load flags? I would suggest to clarify that comment.

Reply:
Will add a comment similar to the one on `prefer_interpreter` that mentions the load flags.

Reply:
Done

Reply:
Ack, thanks for rewording that. IMO very helpful to have these mappings.